MarkBind/markbind
# Fix bs-modal css variables #2353

**What is the purpose of this pull request?**




**Overview of changes:** Fixes #2353 

- Add `.modal` class to `vue-final-modal` component
- Add `display=block` for the `.modal` class to disable display toggling provided by bootstrap
- Downgrade bootswatch dependency to v5.1.3 in sync with current bootstrap version
- Add warning about coupling of bootstrap and bootswatch in dev guide


**Before:**

When using a (non-default) bootswatch theme, bs-modal css variables for modal component are not found, resulting in it not being styled properly:

<img width="1440" alt="Screenshot 2023-08-09 at 11 00 26 PM" src="https://github.com/MarkBind/markbind/assets/87931905/2c7ff0cb-994e-44e0-9652-b6b6a6c245e6">


<em><strong>bootswatch theme used: 'sketchy'</strong></em>

<br>

**After:**

bs-modal CSS variables now registered in modal component:

<img width="1440" alt="Screenshot 2023-08-09 at 10 55 44 PM" src="https://github.com/MarkBind/markbind/assets/87931905/5f56e4eb-295d-41d0-bb40-aa6af07226f2">



**Anything you'd like to highlight/discuss:**


**Testing instructions:**



1. update `site.json` in `packages/cli/test/functional/test_site` to add any non-default bootswatch theme (see https://markbind.org/userGuide/themes.html)
1. cd into `packages/cli/test/functional/test_site` and run `markbind serve -d`
1. navigate to `test_site/testModals.html` in browser and ensure that all modals are rendered correctly in terms of styling

**Proposed commit message: (wrap lines at 72 characters)**

Register bs-modal css variables in Modal component




---

**Checklist:** :ballot_box_with_check:
